app/components/trainees/confirmation/personal_details/view.rb

```diff
@@ -0,0 +1,45 @@
+module Trainees
+  module Confirmation
+    module PersonalDetails
+      class View < GovukComponent::Base
+        attr_accessor :trainee
+
+        def initialize(trainee:)
+          @trainee = trainee
+          @not_provided_copy = I18n.t("components.confirmation.not_provided")
+        end
+
+        def full_name
+          return @not_provided_copy unless trainee.first_names && trainee.last_name
+
+          "#{trainee.first_names} #{trainee.middle_names} #{trainee.last_name}".upcase
```

Comment:
Omitting middle names gives double space...

Are we ok with that?

Reply:
fixed now